### PR TITLE
chore(dependencies): Upgrade to latest versions

### PR DESCRIPTION
**Pull request by Kurt-von-Laven**
Remain on Python 3.9.7, because of an issue encountered running pre-commit on Python 3.10.

### CODE REVIEW

File: poetry.lock [hunk 1]
```diff
@@ -39,11 +27,11 @@ python-versions = ">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*"
 
 [[package]]
 name = "commitizen"
-version = "2.20.0"
+version = "2.23.0"
 description = "Python commitizen client tool"
 category = "dev"
 optional = false
-python-versions = ">=3.6.1,<4.0.0"
+python-versions = ">=3.6.2,<4.0.0"
```
Comment (davisgtaylor) [inline poetry.lock:34]: nit(? did I do that right?) - this upgrade is not listed in the description

Comment (davisgtaylor) [reply poetry.lock:34]: I imagine this is just understood to fall withing "all transitive dependencies"?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -90,12 +90,12 @@ repos:
           - --env
           - LOG_LEVEL=warning
           - --release
-          - v5.0.7
+          - v5.10.0
```
Comment (davisgtaylor) [inline .pre-commit-config.yaml:93]: similar to the last comment  - this update is not referenced within the commit message

**Review by davisgtaylor — COMMENTED** [review]
simple comments but worth answering prior to approval

File: poetry.lock (hunk 1 above)
Comment (Kurt-von-Laven) [reply poetry.lock:34]: In general, you are correct that the modifications to `poetry.lock` typically fall in the transitive dependency category, although the specific line you noted actually details the supported versions of Python for one of our direct dependencies, commitizen. The lock file is managed automatically by Poetry, our Python dependency manager, so I modified `pyproject.toml` and then ran `poetry update` to generate these changes. One way of validating the changes to `poetry.lock` is to repeat that process locally and confirm that the resulting lock files are identical, but our pre-commit hooks actually already validate the lock file in CI, so testing it again locally typically isn't expected of a reviewer.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (Kurt-von-Laven) [reply .pre-commit-config.yaml:93]: The commit subject is "chore(MegaLinter): Upgrade from v5.0.7 to v5.10.0."

Comment (davisgtaylor) [reply .pre-commit-config.yaml:93]: comment redacted

**Review by davisgtaylor — APPROVED** [review]
LGTM

File: poetry.lock (hunk 1 above)
Comment (Kurt-von-Laven) [reply poetry.lock:34]: Yeah, flagging nits vs. other feedback isn’t really a standardized thing, but your way is clear.

**Pull request merged**